language-plutus-core/src/Language/PlutusCore/Evaluation/Machine/Ck.hs

```diff
@@ -28,8 +29,18 @@ import           Language.PlutusCore.Pretty                                 (Pre
 import           Language.PlutusCore.Universe
 import           Language.PlutusCore.View
 
+import           Data.Array
+
 infix 4 |>, <|
 
+builtinNameArities :: Array BuiltinName Int
+builtinNameArities =
+    listArray (minBound, maxBound) $
+        [minBound..maxBound] <&> \name ->
+            withTypedBuiltinName @_ @(Term TyName Name DefaultUni ()) name $
+                \(TypedBuiltinName _ sch) -> countArgs sch
+{-# NOINLINE builtinNameArities #-}  -- Just in case.
```

Comment:
Let's not duplicate this function in the machines. We can extract it into `Constant.Apply`, for example (or maybe there's a better place).

Reply:
:+1:

Reply:
I've move that into `Apply.hs`.  It will also go away if we get rid of static builtins.